core/src/SYCL/Kokkos_SYCL_Instance.cpp

```diff
@@ -160,7 +168,13 @@ void SYCLInternal::finalize() {
 
   // The global_unique_token_locks array is static and should only be
   // deallocated once by the defualt instance
-  if (this == &singleton()) Impl::sycl_global_unique_token_locks(true);
+  if (this == &singleton()) {
+    Impl::sycl_global_unique_token_locks(true);
+#ifdef KOKKOS_IMPL_SYCL_DEVICE_GLOBAL_SUPPORTED
+    desul::Impl::finalize_lock_arrays();
```

Comment:
I would have thought that you need to call these two functions in reverse order of the init call sequence?
Maybe it does not matter?

Reply:
They are independent.